### PR TITLE
Adds voice provider. Currently only Twilio, NotificationCatcher, and custom supported

### PR DESCRIPTION
**Pull request by armiiller**

### CODE REVIEW

**Comment by armiiller** [comment]
What is the acceptance on the tests?

I am getting:
Jest: "global" coverage threshold for statements (100%) not met: 96.41%
Jest: "global" coverage threshold for branches (100%) not met: 94.56%

**Comment by BDav24** [comment]
Hi @armiiller 
Thanks for the PR!
Configuration for coverage is 100% (because it's a small project :)), but this can be fixed later.
I'll try to review the code next week.

**Comment by armiiller** [comment]
@BDav24 - Did you get a chance to check this out?

**Comment by BDav24** [comment]
Sorry, I had a big week.
I'm on it.

File: src/providers/voice/twilio.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      statusCallbackMethod,
+      statusCallbackEvent,
+      sendDigits,
+      machineDetection,
+      machineDetectionTimeout,
+      timeout,
+      recordingChannels,
+      recordingStatusCallback,
+      recordingStatusCallbackEvent
+    } = request
```
Comment (BDav24) [inline src/providers/voice/twilio.js]: All these params seem a bit too specific to Twilio. Do you need any of them (apart from `from`, `to`, and `url`)?

Comment (armiiller) [reply src/providers/voice/twilio.js]: Yes, for the purposes of PagerTree we actually use from, to, url, fallbackUrl, statusCallback, statusCallbackEvent, machineDetection, MachineDetectionTimeout, and timeout.

If for some response its too specific and a another provider is implemented, this type can always follow the flexible  typescript definition like in src/models/provider-email.js

Comment (BDav24) [reply src/providers/voice/twilio.js]: Ok, I removed `statusCallbackMethod`, `recordingChannels`, `recordingStatusCallback`, and `recordingStatusCallbackEvent` for now (not all providers propose the same possibilities, so the simplest, the better :)).

File: src/providers/voice/twilio.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    const form = new FormData()
+    form.append('From', from)
+    form.append('To', to)
+    form.append('Url', url)
+    if (method) form.append('Method', method)
+    if (fallbackUrl) form.append('FallbackUrl', fallbackUrl)
+    if (fallbackMethod) form.append('FallbackMethod', fallbackMethod)
+    if (statusCallback) form.append('StatusCallback', statusCallback)
+    if (statusCallbackMethod) form.append('StatusCallbackMethod', statusCallbackMethod)
+    if (statusCallbackEvent) form.append('StatusCallbackEvent', statusCallbackEvent)
```
Comment (BDav24) [inline src/providers/voice/twilio.js]: The `form-data` lib does not accept arrays, is it ok if I change the code by:
```js
if (statusCallbackEvent) {
  statusCallbackEvent.forEach((event) => form.append('StatusCallbackEvent', event))
}
```
?

File: src/providers/voice/twilio.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    if (sendDigits) form.append('SendDigits', sendDigits)
+    if (machineDetection) form.append('MachineDetection', machineDetection)
```
[the remark this answers is not in the document]
Comment (armiiller) [reply src/providers/voice/twilio.js]: Yes - good catch. That should be changed to a string

File: src/providers/voice/twilio.js (hunk 2 above)
Comment (armiiller) [reply src/providers/voice/twilio.js]: Yes another good call. Looks like the docs for that is here: https://www.twilio.com/docs/api/voice/making-calls#statuscallback-parameter

File: src/providers/voice/twilio.js (hunk 1 above)
Comment (armiiller) [reply src/providers/voice/twilio.js]: Hmmm, for this i respectfully disagree - I think having a flexible typescript definition would be best. I  can see someone else needing this functionality, and why not just have all the options available? If another vendor comes along, we could add another definition.

Comment (BDav24) [reply src/providers/voice/twilio.js]: Well my first priority is to ensure that people can switch their provider easily. Then I agree that "everything should be possible". Here is what I had in mind:

```js
export type VoiceRequestType = RequestMetadataType & {
  from: string,
  to: string,
  url: string,
  method?: string,
  fallbackUrl?: string,
  fallbackMethod?: string,
  statusCallback?: string,
  statusCallbackEvent?: string[],
  sendDigits?: string,
  machineDetection?: string,
  machineDetectionTimeout?: number,
  timeout?: number,
  twilioSpecifc?: {
    statusCallbackMethod?: string,
    recordingChannels?: string,
    recordingStatusCallback?: string,
    recordingStatusCallbackEvent?: string[]
  }
}
```

What do you think?

**Pull request merged**